node_modules/oae-principals/lib/rest.user.js

```diff
@@ -566,6 +567,63 @@ OAE.tenantRouter.on('post', '/api/user/:userId/picture', function(req, res) {
 });
 
 /**
+ * @REST postUserEmailVerify
+ *
+ * Verify an email token
+ *
+ * @Server      tenant
+ * @Method      POST
+ * @Path        /user/email/verify
+ * @FormParam   {string}        email                       The email to verify
+ * @FormParam   {string}        token                       The token to verify
+ * @HttpResponse                200                         The token was valid and the email address for the user that was associated with it is verified
+ * @HttpResponse                400                         A token was not specified
+ * @HttpResponse                404                         The token was not associated with a user
+ */
+OAE.tenantRouter.on('post', '/api/user/email/verify', function(req, res) {
+    PrincipalsAPI.verifyEmail(req.ctx, req.body.email, req.body.token, function(err, user) {
+        if (err) {
+            return res.send(err.code, err.msg);
+        }
+
+        // If the user is anonymous we log them into the system
+        if (!req.ctx.user()) {
+            return AuthenticationUtil.handleLogin('local', user, req, res);
```

Comment:
Errmm. I think we will actually need them to authenticate in order to verify their email address. Not do it anonymously.